packages/jaeger-ui/src/middlewares/index.js

```diff
@@ -45,4 +47,29 @@ export const historyUpdateMiddleware = store => next => action => {
   return next(action);
 };
 
+export const transformOTLPMiddleware = store => next => action => {
+  if (action.type === String([`${loadJsonTraces}_FULFILLED`])) {
+    // Check if action.payload is OTLP and make API call if so
+    // We are allowed to change the action.payload here
+    //
+    if ('resourceSpans' in action.payload) {
```

Comment:
1. please add a unit test
2. did you try to get an official JSON output for OTEL data? E.g. by running OTEL Collector with debugexporter (as described in the ticket). I would suggest saving a trimmed down version of that output as a test fixture file here and loading it in the unit test

Reply:
> 2. did you try to get an official JSON output for OTEL data? E.g. by running OTEL Collector with debugexporter (as described in the ticket). I would suggest saving a trimmed down version of that output as a test fixture file here and loading it in the unit test

Yes I have tested the enter pipeline from uploading OTLP debug expoerted traces to having them show up in the UI....I'll look int adding them as test fixuters

Reply:
This is now handled in index.test.js, a new folder called `fixtures` was created to house these trace examples